domain/ide/workspace/folder_test.go

```diff
@@ -463,25 +465,51 @@ func Test_processResults_ShouldSendAnalyticsToAPI(t *testing.T) {
 		Issues:  []snyk.Issue{mockCodeIssue},
 	}
 
-	engineMock.EXPECT().GetConfiguration().AnyTimes().Return(gafConfig)
-	engineMock.EXPECT().InvokeWithInputAndConfig(localworkflows.WORKFLOWID_REPORT_ANALYTICS, gomock.Any(), gomock.Any()).
-		// this captures the call parameters of the mocked call
-		Do(func(id workflow.Identifier, workflowInputData []workflow.Data, config configuration.Configuration) {
-			require.Equal(t, 1, len(workflowInputData))
-			payloadBytes, ok := workflowInputData[0].GetPayload().([]byte)
-			require.True(t, ok)
-
-			var scanDoneEvent json_schemas.ScanDoneEvent
-			err := json.Unmarshal(payloadBytes, &scanDoneEvent)
+	ic := analytics.NewInstrumentationCollector()
+
+	ua := util.GetUserAgent(gafConfig, config.Version)
+	ic.SetUserAgent(ua)
+	categories := setupCategories(&data, c)
+	ic.SetCategory(categories)
+	ic.SetStage("dev")
+	ic.SetStatus("Success") //or get result status from scan
+	ic.SetInteractionType("Scan done")
+	summary := createTestSummary(&data, c)
+	ic.SetTestSummary(summary)
+	ic.SetType("Analytics")
+
+	entered := make(chan struct{})
+	_, err := engineMock.Register(localworkflows.WORKFLOWID_REPORT_ANALYTICS, workflow.ConfigurationOptionsFromFlagset(pflag.NewFlagSet("", pflag.ContinueOnError)),
+		func(invocation workflow.InvocationContext, workflowInputData []workflow.Data) ([]workflow.Data, error) {
+			actualV2InstrumentationObject, err := analytics.GetV2InstrumentationObject(ic)
+
 			require.NoError(t, err)
-			require.Equal(t, "Snyk Open Source", scanDoneEvent.Data.Attributes.ScanType)
-			require.Equal(t, 1, scanDoneEvent.Data.Attributes.UniqueIssueCount.Medium)
+
+			require.Equal(t, "snyk-ls", actualV2InstrumentationObject.Data.Attributes.Runtime.Application.Name)
+			require.Equal(t, "dev", string(*actualV2InstrumentationObject.Data.Attributes.Interaction.Stage))
+			require.Equal(t, "Success", actualV2InstrumentationObject.Data.Attributes.Interaction.Status)
+			require.Equal(t, "Scan done", actualV2InstrumentationObject.Data.Attributes.Interaction.Type)
+			require.Equal(t, []string{product.ToProductCodename(data.Product), "test"}, *actualV2InstrumentationObject.Data.Attributes.Interaction.Categories)
+			require.Equal(t, "Analytics", actualV2InstrumentationObject.Data.Type)
```

Comment:
If we don't have a unit test for InstrumentationCollector, we should probably test all attributes

Reply:
There is an IntstrumentationCollectorTest in the gaf package.